src/bluesearch/database/article.py

```diff
@@ -102,6 +141,17 @@ def pmc_id(self) -> Optional[str]:
         """
         return None
 
+    @property
+    def arxiv_id(self) -> Optional[str]:
```

Comment:
To be honest I have never liked `Optional[Type]`, as I think it is a bit of a "misnaming" — especially when it's for an argument, it's easy to think of it as an "optional argument", while this is very not what it means at all.
So I would even be more radical and propose to replace _everywhere_ in our our code `Optional[Type]` with `Type | None`.

<details>
  <summary>(Side note)</summary>
I find `str | None` also somehow misleading, as `type(None)` is not `None` but `NoneType` and therefore I would expect `str | NoneType`. The docs of `mypy` say  [here](https://mypy.readthedocs.io/en/stable/kinds_of_types.html#optional-types-and-the-none-type) that it's because it "is shorter and reads better", and [this issue](https://github.com/python/mypy/issues/11288) adds that "`NoneType` is a mangled up mess in the type realm and should never be used in a typing context".
</details>

TL;DR I used `Optional[str]` here just because I have seen it being used more often than `str | None` in our codebase. But I think `Optional` is a terrible naming choice, and if everyone here agrees  I can replace `Optional[T]` with `T | None` everywhere.

Reply:
Yeh, I agree. I would suggest creating a separate PR/issue for the replacement in the entire codebase.

Reply:
Switched to `str | None` in 80c2053.

Opened https://github.com/BlueBrain/Search/pull/537 to use `X | Y` consistently in our codebase.